packages/jest/src/plugins/plugin.ts

```diff
@@ -208,90 +199,172 @@ async function buildJestTargets(
   const cache = (target.cache = true);
   const inputs = (target.inputs = getInputs(
     namedInputs,
-    rawConfig,
+    rawConfig.preset,
     projectRoot,
-    context.workspaceRoot
+    context.workspaceRoot,
+    options.disableJestRuntime
   ));
-  const outputs = (target.outputs = getOutputs(projectRoot, config, context));
 
   let metadata: ProjectConfiguration['metadata'];
-  if (options?.ciTargetName) {
-    // nx-ignore-next-line
-    const { default: Runtime } = requireJestUtil<typeof import('jest-runtime')>(
-      'jest-runtime',
-      projectRoot,
-      context.workspaceRoot
-    );
 
-    const jestContext = await Runtime.createContext(config.projectConfig, {
-      maxWorkers: 1,
-      watchman: false,
-    });
+  const groupName = 'E2E (CI)';
 
-    const jest = require(resolveJestPath(
+  if (options.disableJestRuntime) {
+    const outputs = (target.outputs = getOutputs(
       projectRoot,
-      context.workspaceRoot
-    )) as typeof import('jest');
-    const source = new jest.SearchSource(jestContext);
-
-    const jestVersion = getInstalledJestMajorVersion()!;
-    const specs =
-      jestVersion >= 30
-        ? // @ts-expect-error Jest 30+ expects the project config as the second argument
-          await source.getTestPaths(config.globalConfig, config.projectConfig)
-        : await source.getTestPaths(config.globalConfig);
-
-    const testPaths = new Set(specs.tests.map(({ path }) => path));
-
-    if (testPaths.size > 0) {
-      const groupName = 'E2E (CI)';
-      const targetGroup = [];
-      metadata = {
-        targetGroups: {
-          [groupName]: targetGroup,
-        },
-      };
-      const dependsOn: string[] = [];
-
-      targets[options.ciTargetName] = {
-        executor: 'nx:noop',
-        cache: true,
-        inputs,
-        outputs,
-        dependsOn,
-        metadata: {
-          technologies: ['jest'],
-          description: 'Run Jest Tests in CI',
-          nonAtomizedTarget: options.targetName,
-          help: {
-            command: `${pmc.exec} jest --help`,
-            example: {
-              options: {
-                coverage: true,
+      rawConfig.coverageDirectory
+        ? join(context.workspaceRoot, projectRoot, rawConfig.coverageDirectory)
+        : undefined,
+      undefined,
+      context
+    ));
+
+    if (options?.ciTargetName) {
```

Comment:
nit: we can probably avoid a lot of duplication building the target by moving the handling of the `disableJestRuntime` option into the different methods, but that can be a follow-up cleanup.

Reply:
I think it's a rule-of-three situation. Duplicating once is fine, a third time is when you extract.